classes/Product.php

```diff
@@ -764,10 +762,13 @@ public function __construct($id_product = null, $full = false, $id_lang = null,
     public function getFieldsShop()
     {
         $fields = parent::getFieldsShop();
-        if (null === $this->update_fields || (!empty($this->update_fields['price']) && !empty($this->update_fields['unit_price']))) {
-            if ($this->unit_price !== null) {
-                $fields['unit_price_ratio'] = (float) $this->unit_price > 0 ? $this->price / $this->unit_price : 0;
-            }
+
+        // In case unit_price is not in fields we can try and fallback with unit_price_ratio
+        if (null !== $this->unit_price_ratio && empty($fields['unit_price']) && (
+            null === $this->update_fields ||
+            (!empty($this->update_fields['price']) && !empty($this->update_fields['unit_price_ratio']) && empty($this->update_fields['unit_price']))
+        )) {
+            $fields['unit_price'] = (float) $this->unit_price_ratio > 0 ? $this->price / $this->unit_price_ratio : 0;
```

Comment:
Shouldn't this be done using decimal to avoid loss of precision?

Reply:
Maybe but honestly I kept this code to avoid changing too many things I see no reason why this could happen

Reply:
Actually, this code is now useless so I'm gonna clean it because it's now useless